independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/mapping/RequestMapper.java

```diff
@@ -53,14 +69,16 @@ public RequestMatch<T> map(String path) {
     }
 
     @SuppressWarnings({ "rawtypes", "unchecked" })
-    private RequestMatch<T> mapFromPathMatcher(String path, PathMatcher.PathMatch<ArrayList<RequestPath<T>>> initialMatch) {
+    private List<RequestMatch<T>> mapFromPathMatcher(String path,
+            PathMatcher.PathMatch<ArrayList<RequestPath<T>>> initialMatch) {
         var value = initialMatch.getValue();
         if (initialMatch.getValue() == null) {
-            return null;
+            return Collections.emptyList();
         }
         int pathLength = path.length();
-        for (int index = 0; index < ((List<RequestPath<T>>) value).size(); index++) {
-            RequestPath<T> potentialMatch = ((List<RequestPath<T>>) value).get(index);
+        List<RequestMatch<T>> matches = new ArrayList<>();
```

Comment:
As explained at https://github.com/quarkusio/quarkus/pull/47386/files#r2047618963; why having the list at all if the caller just need the top element?

Reply:
done